src/exports.cc

```diff
@@ -767,18 +767,46 @@ Word wasi_unstable_fd_fdstat_get(void *raw_context, Word fd, Word statOut) {
 }
 
 // __wasi_errno_t __wasi_environ_get(char **environ, char *environ_buf);
-Word wasi_unstable_environ_get(void *, Word, Word) {
+Word wasi_unstable_environ_get(void *raw_context, Word environ_array_ptr, Word environ_buf) {
+  auto context = WASM_CONTEXT(raw_context);
+  auto word_size = context->wasmVm()->getWordSize();
+  auto &envs = context->wasm()->envs();
+  for (auto e : envs) {
+    if (!context->wasmVm()->setWord(environ_array_ptr, environ_buf)) {
+      return 21; // __WASI_EFAULT
+    }
+
+    std::string data;
+    data.reserve(e.first.size() + e.second.size() + 2);
+    data.append(e.first);
+    data.push_back('=');
```

Comment:
Any reason for using both `append` and `push_back` here? Shouldn't we be consistent and use `append`?

Reply:
string::append requires const pointer to chars so cannot pass the char literal. Is there any workaround?

Reply:
Passing those values as strings and not chars (i.e. `data.append("=")` and `data.append("\0")`) should work.

Reply:
Thanks!

Reply:
Nit: in C/C++ `\0` is generally used to indicate `NULL` terminator, not `\x0` (even though they are technically the same).